holoviews/element/raster.py

```diff
@@ -610,7 +610,8 @@ def __init__(self, data, kdims=None, vdims=None, **params):
             arrays = [(im.data - r[0]) / (r[1] - r[0]) for r,im in zip(ranges, images)]
             data = np.dstack(arrays)
         if vdims is None:
-            vdims = list(self.vdims)
+            # Same as the class variables, put here to secure the class variable is not used
+            vdims = [Dimension(c, range=(0,1)) for c in "RGB"]
```

Comment:
I'm quite curious why a shallow copy of `vdims` is made on this line on an line 616, but it's too late to dive into an archeology session!

I don't think this change is correct. `vdims` seems to accept a list of len 3 or 4 (A?). Also sub classes (e.g. https://github.com/holoviz/holoviews/blob/8ff79768743aff865ed5d42246d67c61042abefb/holoviews/element/raster.py#L677) can have different Dimensions without having any special handling in their __init__. For instance (after the change):

```python
>>> import holoviews as hv
>>> hv.HSV([]).vdims
[Dimension('R'), Dimension('G'), Dimension('B')]
```

Maybe a deepcopy of `self.vdims` would be better?

Reply:
You are right. I originally used `deepcopy` but felt it was overkill. I will create a new PR with the correct change and a unit test.